### PR TITLE
18 method for price histories

### PR DESCRIPTION
**Pull request by peterwhiteenjine**

### CODE REVIEW

File: mcos/observation_simulator.py [hunk 1]
```diff
@@ -38,3 +41,15 @@ def __init__(self, mu: np.array, cov: np.array, n_observations: int):
...
+def convert_price_history(df: pd.DataFrame):
+    """
+    converts a price history dataframe into expected returns and covariance
+     :param df: Dataframe of price histories indexed by data
+     @return
+    """
+    # Calculate expected returns and sample covariance
+    mu = expected_returns.mean_historical_return(df)
+    cov = risk_models.sample_cov(df)
+    return mu, cov
```
Comment (andystanley) [inline mcos/observation_simulator.py:55]: Should we move this method should be moved to a `utils.py` file? If I was looking for this method, I wouldn't expect to find it in `observation_simulator.py`.

File: tests/test_mcos.py [hunk 2]
```diff
@@ -108,9 +109,20 @@ def test_simulate_observations(simulator, estimator, transformers, expected_mean
...
+
+
+def test_convert_price_history():
+    df = pd.read_csv('test/stock_prices.csv', parse_dates=True, index_col='date')
+    mu, cov = convert_price_history(df)
+
+    assert mu["GOOG"], 0.26770283812412754
+    assert mu['AAPL'], 0.36378640487631986
+    assert cov['GOOG']['AAPL'], 0.04620227917174632
+    assert cov['GE']['FB'], 0.014789747995647461
```
Comment (andystanley) [inline tests/test_mcos.py:128]: Similarly, this test should probably be in a different file

File: mcos/observation_simulator.py [hunk 3]
```diff
@@ -38,3 +41,15 @@ def __init__(self, mu: np.array, cov: np.array, n_observations: int):
     def simulate(self) -> (np.array, np.array):
         x = np.random.multivariate_normal(self.mu.flatten(), self.cov, size=self.n_observations)
         return x.mean(axis=0).reshape(-1, 1), np.cov(x, rowvar=False)
+
+
+def convert_price_history(df: pd.DataFrame):
+    """
+    converts a price history dataframe into expected returns and covariance
+     :param df: Dataframe of price histories indexed by data
+     @return
```
Comment (ochrist2) [inline mcos/observation_simulator.py:50]: can we add what this returns?

File: mcos/mcos.py [hunk 4]
```diff
@@ -37,3 +39,24 @@ def simulate_optimizations(
             'stdev': np.std(error_estimates[optimizer.name])
         } for optimizer in optimizers
     ]).set_index('optimizer')
+
+
+def simulate_optimization_from_price_history(
+        price_history: pd.DataFrame,
+        observation_name: str,
```
Comment (moneygeek) [inline mcos/mcos.py:46]: can we rename this to `simulator_name`?

File: mcos/mcos.py [hunk 5]
```diff
@@ -37,3 +39,24 @@ def simulate_optimizations(
...
+        n_observations:int,
+        n_sims: int,
+        optimizers: List[AbstractOptimizer],
+        error_estimator: AbstractErrorEstimator,
+        covariance_transformers: List[AbstractCovarianceTransformer]):
+
+    mu, cov = convert_price_history(price_history)
+
+    if observation_name.lower() == "mucovledoitwolfobservationsimulator":
```
Comment (moneygeek) [inline mcos/mcos.py:55]: Can we rename this to `mucovledoitwolf`? Truncate the other simulator's name as well.

Comment (moneygeek) [reply mcos/mcos.py:55]: Could you also provide the list of valid simulator names in the README?

**Pull request merged**